mantidproject/mantid
# Muon Results table not plotting correctly

The results table produced by muon analysis was having numeric x-axis interpreted as strings (see image in issue). This PR allows the table to have numeric columns were applicable. 

**To test:**
to set up the test:

1. Open muon analysis
2. Select the EMU instrument
3.  Load 20717


The first test

1.  Go to the settings tab and make sure that multiple fitting is not selected
2. Go to the Data analysis tab and add a fitting function. 
3. Do a fit
4. Go to the Home tab and change the group/pair (should have long, fwd and bkwd).
5. Go back to data analysis and do another fit
6. Go to the home tab and pick the remaining option for the group/pair
7. Go to the data analysis tab and run a fit
8. Create the results table.
9. Make sure you can tell which row matches with each of the groups

The second test:
1. Go to the settings tab and make sure that multiple fitting is not selected
2. Go to the Data analysis tab and add a fitting function. 
3. Go to the fit menu and select 'sequential fit'
4. In the runs box put '20717-26, 20730-33'
5. Click run
6. Once its complete close the tab and go to the results table tab
7. In the middle page are some buttons, select sequential.
8. In the top box select sample_Temp
9. Click create table.
10. Plot (from the table) the value of a parameter with respect to sample temp. Notice that in the table sample temp is not in ascending order.  
11. It works if the plot looks normal (i.e. the x axis does not suddenly decrease in value)

Third test:
1. Go to the settings tab and make sure that multiple fitting is selected
2. Go to the Data analysis tab and add a fitting function. 
3. Make sure 'all groups' is selected in the group/pair drop down menu
4. Share one of the parameters (e.g. freq)
5. Run a fit
6. Go to the home tab and load the next data set. Then repeat steps 2 to 5. Do this a couple time.
7. Go to the results table tab and select the multiple button and generate a table.
8. For the non-shared parameters there should be 2 occurrence of each (e.g. f0.A and f1.A). 
9. For shared parameters should only be in the table once. 



Fixes #21132. 

**Release Notes** 


---
In the release notes
#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
